plugins/server-stats/frontend/public/javascripts/countly.models.js

```diff
@@ -85,6 +112,447 @@
+        var padding = 3;
+        var xLabelHeight = 30;
+        var yLabelWidth = 80;
+        var borderWidth = 1;
+        var duration = 0;
+
+
+
+        var chart = d3.select('#chart').append('svg')
+            .attr('width', width + margin.left + margin.right)
+            .attr('height', height + margin.top + margin.bottom)
+            .append('g')
+            .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+
+        var border = chart.append('rect')
+            .attr('id', 'tod-chart-area')
+            .attr('x', yLabelWidth)
+            .attr('y', xLabelHeight)
+            .style('fill-opacity', 0)
+            .style('stroke', '#D0D0D0')
+            .style('stroke-width', borderWidth)
+            .style('shape-rendering', 'crispEdges');
+
+
+        loadPunchCard(dpPunchData);
+
+        /**
+         * Load d3 punch chard
+         * @param {object} punchCardData | Chart data
+         */
+        function loadPunchCard(punchCardData) {
+
+            var labelsX = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23"];
+            var labelMapping = {
+                "0": jQuery.i18n.map['times-of-day.sunday'],
+                "1": jQuery.i18n.map['times-of-day.monday'],
+                "2": jQuery.i18n.map['times-of-day.tuesday'],
+                "3": jQuery.i18n.map['times-of-day.wednesday'],
+                "4": jQuery.i18n.map['times-of-day.thursday'],
+                "5": jQuery.i18n.map['times-of-day.friday'],
+                "6": jQuery.i18n.map['times-of-day.saturday']
+            };
+
+            var data = [];
+            for (var i = 0; i < punchCardData.length; i++) {
+                data.push({
+                    label: labelMapping[i],
+                    values: punchCardData[i],
+                });
+            }
+
+            var sunday = data[0];
+            data = data.splice(1, 7);
+            data.push(sunday);
+
+            update(data, labelsX);
+
+            $('#mouseOverRectEvent').tooltipster({
+                animationDuration: 0,
+                delay: 100,
+                theme: 'tooltipster-borderless',
+                trigger: 'custom',
+                triggerOpen: {
+                    mouseenter: true,
+                    touchstart: true
+                },
+                triggerClose: {
+                    mouseleave: true,
+                    touchleave: true
+                },
+                interactive: true,
+                contentAsHTML: true
+            });
+
+        }
+
+        /**
+         * Convert hexadecimal to RGB value
+         * @param {string} hex value for color
+         * @returns {string} converted rgb value of hex color
+         */
+        function hexToRgb(hex) {
+            hex = hex.substr(1, 6);
+            var bigint = parseInt(hex, 16);
+            var r = (bigint >> 16) & 255;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unexpected use of '&'.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5232873)